### PR TITLE
Integration2 t6f2 77 reviews dao

### PR DESCRIPTION
**Pull request by bovornc**
Finished models/schemas/daos/interfaces for the reviews backend component. Let me know how it looks!

### CODE REVIEW

File: daos/ReviewDao.ts [hunk 1]
```diff
@@ -29,15 +29,15 @@ export default class ReviewDao implements ReviewDaoI {
...
 
     public async deleteReview(reviewID: string): Promise<any> {
-        return await TuitModel.deleteOne({_id: reviewID});
+        return await ReviewModel.deleteOne({_id: reviewID});
     }
 
-    public async updateReview(reviewID: string, review: Review): Promise<any> {
-        return Promise.resolve(undefined);
+    public async updateReview(reviewID: string, review: Tuit): Promise<any> {
+        return ReviewModel.updateOne({_id: reviewID, review: review});
```
Comment (nala-he) [inline daos/ReviewDao.ts]: For updateReview line 41, I think it should be ReviewModel.updateOne({_id: reviewID, {$set: review}}). I am not sure though, they just look differently from my updateTuit function. Can Macee and Yena provide some feedback here?

File: daos/ReviewDao.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import Tuit from "../models/Tuit";
+import Review from "../models/Review";
+import ReviewModel from "../mongoose/ReviewModel";
+import ReviewDaoI from "../interfaces/ReviewDaoI";
+
+export default class ReviewDao implements ReviewDaoI {
```
Comment (MaceeQi) [inline daos/ReviewDao.ts:15]: Do you mind adding JSDoc for each of the files?

File: daos/ReviewDao.ts (hunk 1 above)
Comment (MaceeQi) [reply daos/ReviewDao.ts:41]: I'm not too sure either, I've also always used $set

File: interfaces/ReviewDaoI.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import Tuit from "../models/Tuit";
+import Review from "../models/Review";
+
+export default interface ReviewDaoI {
```
Comment (MaceeQi) [inline interfaces/ReviewDaoI.ts:7]: JSDoc please!

File: models/Review.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import Tuit from "./Tuit";
+import Restaurant from "./Restaurant";
+import User from "./User";
+
+export default interface Review {
```
Comment (MaceeQi) [inline models/Review.ts:13]: JSDoc

File: mongoose/ReviewModel.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import mongoose from "mongoose";
+import ReviewSchema from "./ReviewSchema";
+
+const ReviewModel = mongoose.model("ReviewModel", ReviewSchema);
```
Comment (MaceeQi) [inline mongoose/ReviewModel.ts:7]: JSDoc

File: mongoose/ReviewSchema.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import mongoose from "mongoose";
+import Review from "../models/Review";
+
+const ReviewSchema = new mongoose.Schema<Review>({
```
Comment (MaceeQi) [inline mongoose/ReviewSchema.ts:12]: JSDoc

**Review by MaceeQi — COMMENTED** [review]
Could you add JSDoc to each of the files? Other than that, everything looks good!

File: daos/ReviewDao.ts (hunk 1 above)
Comment (MaceeQi) [reply daos/ReviewDao.ts]: I'm not too sure either, I've also always used $set as well

**Comment by bovornc** [comment]
Made updates based on feedback/conversations!

**Review by nala-he — APPROVED** [review]
I think it looks good!

**Review by MaceeQi — APPROVED** [review]
Looks good! Thanks for the changes : )

**Pull request merged**